packages/components/src/components/vertical-tab/vertical-tab.jsx

```diff
@@ -28,82 +41,73 @@ class VerticalTab extends React.Component {
             index = typeof selected_index === 'object' ? list.indexOf(selected_index) : selected_index;
         }
 
-        this.setState({ vertical_tab_index: index });
+        setCurrTabIndex(index);
 
-        if (typeof this.props.setVerticalTabIndex === 'function') {
-            this.props.setVerticalTabIndex(index);
+        if (typeof setVerticalTabIndex === 'function') {
+            setVerticalTabIndex(index);
         }
     };
 
-    changeSelected = e => {
-        this.setSelectedIndex({
-            list: this.props.list,
+    const changeSelected = e => {
+        setSelectedIndex({
+            list: list,
             selected_index: e,
         });
     };
 
-    componentDidMount() {
-        this.setSelectedIndex(this.props);
-    }
-
-    componentDidUpdate(prevProps) {
-        if (
-            this.props.list.length !== prevProps.list.length ||
-            this.props.vertical_tab_index !== prevProps.vertical_tab_index
-        ) {
-            this.setSelectedIndex({
-                current_path: this.props.current_path,
-                list: this.props.list,
-                is_routed: this.props.is_routed,
-            });
-        }
-    }
+    React.useEffect(() => {
+        setSelectedIndex({
+            current_path: current_path,
+            list: list,
+            is_routed: is_routed,
+        });
+    }, [vertical_tab_index]);
```

Comment:
Should we add `list` as a dependency here?

Reply:
I was testing whether `list` is a dependency or not as it was in the refactored code it wasn't changing. I re-checked again the `list` was changing as new items being appended so I added it as a dependency 👍🏻